W-11725381-Update-configure-permissions-CP (#58)
mulesoft/docs-code-builder: modules/ROOT/pages/configure-permissions.adoc

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

> You access Anypoint Code Builder using your Anypoint Platform account.

Reviewer: something along these lines just to indicate that this is just about web access.
```suggestion
Access Anypoint Code Builder on the web through your Anypoint Platform account.
```

You manage user access either by granting
xref:access-management::users.adoc#grant-user-permissions[permissions] to users individually
or by using
xref:access-management::teams.adoc[teams].

== Before You Begin

* Accept Anypoint Code Builder beta terms and conditions:
+
--
* For Anypoint Code Builder, your organization administrator must accept Anypoint Code Builder beta terms and conditions.

> * For Anypoint Code Builder for Desktop, you must accept Anypoint Code Builder beta terms and conditions.

Reviewer: We can remove the desktop bullet and reword this section as needed since the permission is only for ACB Web (see to Simone's response mentioned above).

```suggestion
```

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

> You access Anypoint Code Builder using your Anypoint Platform account.

Reviewer: Let me get for info about the sentence above. 

You can access ACB from the desktop IDE without having an Anypoint Platform account. But there may be things you can't do without having an account and that permission. I am double checking that the user/team permissions are required for the desktop user. I am like 98% certain they are, but wan't to be at 100% or feel like 😵‍💫

Reviewer: Ok, Simone says the permission is only needed for Web, not Desktop. Strange but true. I guess it is simply about access to the ACB code builder web IDE.

Reviewer: so... something along these lines just to indicate that this is just about web access:

```suggestion
Access Anypoint Code Builder on the web through your Anypoint Platform account.
```

You manage user access either by granting
xref:access-management::users.adoc#grant-user-permissions[permissions] to users individually
or by using
xref:access-management::teams.adoc[teams].

== Before You Begin

* Ensure that you have access to the private beta release of Anypoint Code Builder.

> * Verify that Anypoint Code Builder beta terms and conditions are accepted:

Reviewer: I don't know that they specifically "verify" that the terms are accepted if they're not an org admin, meaning I don't know what happens if you don't have the permission and try to open ACB from Anypoint Platform on the web. 

It's more like accepting them is just required. If you are using the web IDE, the org admin has to accept the terms and conditions.

If the desktop IDE, you can accept them even if you are not an org admin.

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

//LOGO (web, desktop, or both)
include::partial$acb-ide-logos.adoc[tags="web-only"]

Access the Anypoint Code Builder web-based IDE through your Anypoint Platform account.
You manage user access either by granting
xref:access-management::users.adoc#grant-user-permissions[permissions] to users individually
or by using
xref:access-management::teams.adoc[teams].

== Before You Begin

* Accept Anypoint Code Builder beta terms and conditions.
+
For Anypoint Code Builder on the web, your organization administrator must accept Anypoint Code Builder beta terms and conditions. 
See xref:accept-terms-and-conditions.adoc[Accept Terms and Conditions].

> == Grant Anypoint Code Builder Permissions to Users

Reviewer: Should we say "Permissions to" or "the ... Permission to" given it's one permission. Does it matter. I barely noticed it, tbh.

To grant Anypoint Code Builder Developer permission to an individual user:

. Log in to https://www.anypoint.mulesoft.com/login/[Anypoint Platform^] and navigate to *Access Management*.
. Follow the steps in xref:access-management:users.adoc#grant-user-permissions[Grant Permissions to Users].
+
Assign the Anypoint Code Builder Developer permission to the user.

// image::access-management-select-access.png[“*Anypoint Code Builder* tab, *Select Access* drop-down menu, and *Anypoint Code Builder Developer* option highlighted”]

The user can now develop using Anypoint Code Builder.

> == Grant Anypoint Code Builder Permissions to Teams

Reviewer: same question as above: "Permissions" plural ok?

> Anypoint Code Builder is a web-based IDE that you can access from your Anypoint Platform account. As an organization administrator, you can enable your users to access Anypoint Code Builder.

Reviewer: ```suggestion
Anypoint Code Builder provides an IDE that you can access from your Anypoint Platform account. As an organization administrator, you can enable your users to access Anypoint Code Builder.
```

hold off on committing or merging for a bit while I am double checking that the permission is needed for both IDEs, web and desktop; waiting to get 100% confirmation just to be on the safe side

Reviewer 2: I changed this s/d

== Prerequisites

> * Ensure that you have access to the private beta release of Anypoint Code Builder.

Reviewer: We're not in private beta anymore. Web is already in open beta, and Desktop will be in open beta at release time. And both are accessible without the need for special permission.

Reviewer 2: Wth, I changed that heading to conform to style guidelines!

> * Ensure that you have access to the private beta release of Anypoint Code Builder.

Reviewer: ```suggestion
```

> * Verify that your organization administrator accepted Anypoint Code Builder beta terms and conditions. +

Reviewer: TBD (I've sent out a request for more info): I'm not sure if an org admin has to accept the terms for both web and desktop IDEs. I was able to accept the terms from desktop. I also accepted them on the web, but I owned the instance, so need more info about this.

Not sure "Verify" conveys the intent. You'll either be able to access or not. If not, then you'll apparently need to org admin to accept the terms. It's a precondition of using the IDE.

Author: According to Simone Geib's words:
"No, on desktop we don't have the same control as on web. For web IDE, it is true, only the org admin can accept the T&Cs. For desktop, it's the individual user".
So I reworded the bullet.

. Log in to Anypoint Platform and navigate to *Access Management*:
+
image::select-access-management.png[“*Access Management* highlighted in the Anypoint Platform navigation panel”]
. On the navigation bar, select *Users*:
+
image::select-users-access-management.png[“*Users* option highlighted in the *Access Management*”]

> . Click the user name of the user to which you assign permissions:

Reviewer 2: ```suggestion
. Click the user name to assign permissions to:
```

Author: OK! Thanks!

+
image::select-username-access-management.png[“User name highlighted in the *Organization Users* tab”]
. Navigate to the *Anypoint Code Builder* tab, click the *Select Access* drop-down, and select *Anypoint Code Builder Developer*.
+
image::access-management-select-access.png[“*Anypoint Code Builder* tab, *Select Access* drop-down menu, and *Anypoint Code Builder Developer* option highlighted”]
. Click the plus sign icon to add permission to this user:
+
image::blue-icon-add-access-management.png["Plus sign icon highlighted"]

Your user can now develop using Anypoint Code Builder.

== Grant Teams Permissions to Access Anypoint Code Builder

To grant permissions to use Anypoint Code Builder using Teams:

. Log in to https://www.anypoint.mulesoft.com/login/[Anypoint Platform^] and navigate to *Access Management*:
+
image::select-access-management.png[“*Access Management* highlighted in the Anypoint Platform navigation panel”]
. On the navigation panel, select *Users*:
+
image::select-users-access-management.png[“*Users* option highlighted in the *Access Management*” ]

> . Click the username of the user to which you assign permissions:

Reviewer 2: ```suggestion
. Click the user name to assign permissions to:
```

Reviewer 2: reuse?

+
image::select-username-access-management.png[“User name highlighted in the *Organization Users* tab.”]

> . Navigate to the *Anypoint Code Builder* tab, click the *Select Access* drop-down menu, and select *Anypoint Code Builder Developer*.

Reviewer 2: ```suggestion
. On the *Anypoint Code Builder* tab, select *Select Access* > *Anypoint Code Builder Developer*.
```

+

> image::access-management-select-access.png[“*Anypoint Code Builder* tab, *Select Access* drop-down menu, and *Anypoint Code Builder Developer* option highlighted”]

Reviewer 2: ```suggestion
image::access-management-select-access.png[“*Anypoint Code Builder* tab, *Select Access* menu, and *Anypoint Code Builder Developer* option”]
```

. Click the plus sign icon to add the selected permission to this user:
+
image::blue-icon-add-access-management.png[“Plus sign icon highlighted”]

The user can now develop using Anypoint Code Builder.

> == Grant Teams Permissions to Access Anypoint Code Builder

Reviewer 2: Why are there two topics with the same heading? See line 13.

What is the difference between the two topics?
Can we just give pointers to the AIM docs instead of reproducing them here?

Author: The difference is that the first steps are for users and the latter are for teams:
== Grant **_Users_** Permissions to Access Anypoint Code Builder
== Grant **_Teams_** Permissions to Access Anypoint Code Builder

Reviewer 2: There were three sections. One for Users and two for Teams. I simplified and removed the additional Teams section.

To grant permissions to use Anypoint Code Builder using Teams:

. Log in to https://www.anypoint.mulesoft.com/login/[Anypoint Platform^] and navigate to *Access Management*:
+
image::select-access-management.png[“*Access Management* highlighted in the Anypoint Platform navigation panel.”]

> . Optionally, if you see the message, click *Try New Features* on the navigation panel.

Reviewer: "if you see the message," 

what message?

Reviewer: is it optional if you see that message? Or is it something you have to do if you see the message?

Reviewer: I had the Try New Features button but no message. Wondering if it exists anymore. Or if it exists only under certain conditions.

Reviewer 2: @dukesphere - I think you still need to opt in to Teams:
https://docs.mulesoft.com/access-management/teams#opt-in-to-teams

So maybe something like "If you haven't yet opted in to Teams, click *Try New Features*.

We should link to AIM for the details, regardless.

Author: That's true! The message appears only once, as long as you did not opt-in to the new features before.
You can opt-in to the new features even if you're not part of the ACB beta.

+
image::try-new-features-button.png[“*Try New Features* button highlighted in the Access Management navigation panel.”]
. Click *Start Using These Features* to enable access to the *Teams* feature.
+
image::start-using-these-features.png[“*Start Using These Features* button highlighted in the pop-up window”]

> . Click *Teams* on the navigation panel and *Create Teams*.

Reviewer: This procedure is explaining how to set up teams _in addition to_ setting the ACB permission for a team. I'd suggest a different approach to these steps to focus on the main objective of this page, which is to set the permission for the team, pointing to team setup info elsewhere in the docs for reference to avoid rewriting existing doc and to focus on the purpose of this page. 

It's ok to assume that they can figure out how to add the permission based on the on-screen text.

Reviewer: And better yet, this is really how we should be writing this sort of doc so that we don't create unnecessary maintenance issues for ourselves with cross-product docs:

https://docs.mulesoft.com/mq/mq-access-management#use-teams-to-assign-permissions

Note it's done in 2 steps with a reference for the permissions. In our case though there's only one permission so the whole thing could be written in 2 steps.

Reviewer: Same goes for permissions on the user.

Reviewer: @Crispy-Salesforce and @hannanelson (Hanna has agreed to review this PR.)

Author: Sorry! Do you mean we should create another title to let the reader know how to create a team?
Apart from that, we should have another topic granting permission to such a team?

+
image::teams-create-teams.png[“*Teams* option and *Create team* button highlighted”]
. In the *Create Team* window, type the name of your new team, select the parent team from which permissions are inherited, and click *Create*.
+
image::create-team-window.png[“*Name* and *Parent team* fields highlighted in the *Create Team* pop-up window”]
. Click the name of the team you created and *Add permissions*.
+
image::add-permissions.png[“*Add permissions* and *Parent team* fields highlighted in the *Create Team* pop-up window”]
. Select the *Anypoint Code Builder Developer* option in the *Permissions* list and click *Next*. 
+
image::acb-developer-permission.png[“*Anypoint Code Builder Developer* permission highlighted in the permission page”]

> . Select the business groups where you add the selected permissions and click *Next*.

Reviewer: Do we explain what business groups are anywhere? If so, let's add an xref to it in the See Also info.

Reviewer 2: There's a section on business groups here: https://docs.mulesoft.com/anypoint-code-builder/business-group-support

Reviewer 2: The AIM business groups section: https://docs.mulesoft.com/access-management/business-groups
